calico_node/startup.py

```diff
@@ -266,7 +266,11 @@ def main():
         sys.exit(1)
 
     as_num = os.getenv("AS")
-    as_num = as_num or None
+    if os.getenv("USE_HOST_AS") == "true":
```

Comment:
I think we could do this without introducing a new configuration option:

```python
as_num_env = os.getenv("AS")
if as_num_env:
  as_num = as_num_env
else:
  as_num = client.get_host_as(hostname)
```

The behavior is I think a little nicer:
- If "AS" is defined, use that value.  Otherwise,
- If the hostname has been configured with an AS, use that value.  Otherwise,
- No AS is used